### PR TITLE
feat: Open button checks if folder or item has loadable files

### PR DESCRIPTION
**Pull request by PaulHax**
Adds 2 endpoints:
- GET /item/:id/volview_loadable
- GET /folder/:id/volview_loadable

### CODE REVIEW

File: girder_volview/__init__.py [hunk 1]
```diff
@@ -49,6 +50,47 @@
...
+    Description(
+        "Check if folder has files VolView can load.  If so, return {loadable:true}."
+    )
+    .modelParam("folderId", model=Folder, level=AccessType.READ)
+    .produces(["application/json"])
+    .errorResponse("ID was invalid.")
+    .errorResponse("Read access was denied for the folder.", 403)
+)
+def volViewLoadableFolder(self, folder):
+    files = Folder().fileList(folder, subpath=False, data=False)
```
Comment (PaulHax) [inline girder_volview/__init__.py:89]: Is this using the appropriate database aggregation?

Comment (manthey) [reply girder_volview/__init__.py:89]: Sadly it isn't, but that should be changed in Girder.  This is a perfectly good approach.

File: girder_volview/utils.py [hunk 2]
```diff
@@ -90,15 +90,21 @@ def isSessionFile(file):
...
-    if isSessionFile(file):
-        return False
+def isLoadableFile(file):
     name = file.get("name")
     knownExtension = name.endswith(LOADABLE_EXTENSIONS)
+    if knownExtension:
+        return True
     mimeType = file.get("mimeType")
     # mimeType can be None when imported via S3 asset store
     knownMime = mimeType in LOADABLE_MIMES or mimeType is None
```
Comment (manthey) [inline girder_volview/utils.py:100]: I think this is overly generous; mime type is often not set even on uploaded files if they don't return something appropriate to the python filetype module.  Do we expect unusually named files that also have no mime type?

**Comment by manthey** [comment]
In trying this out, I still see the open in volview button a lot more than I think makes sense.  Specifically, we are matching on unset mime types and on `application/octet-stream`, which covers almost everything that isn't known.  Further, the tiff format can be read by either histomicstk OR volview.  Maybe we should only show the volview button if these tiff items are not marked as large images (which is the trigger to allow them to be opened in histomicstk).

File: girder_volview/utils.py (hunk 2 above)
Comment (PaulHax) [reply girder_volview/utils.py]: Sounds good.  Now `None` is not known.

**Comment by PaulHax** [comment]
Excluding Large Image TIFFs and DICOM SMs now (with another DB hit per file to load their Item)

File: girder_volview/utils.py [hunk 3]
```diff
@@ -76,26 +77,38 @@
...
 
 def isLoadableFile(file):
-    name = file.get("name")
-    knownExtension = name.endswith(LOADABLE_EXTENSIONS)
-    if knownExtension:
+    if isTiffFile(file) or isDicomFile(file):
+        item = Item().load(file.get("itemId"), level=AccessType.READ)
+        if isTiffFile(file):
+            return not item.get("largeImage")
+        if item.get("meta", {}).get("dicom", {}).get("Modality", "") is "SM":
```
Comment (manthey) [inline girder_volview/utils.py:105]: You probably what this to be `==` not `is`.

Comment (PaulHax) [reply girder_volview/utils.py:105]: Fixed.

**Review by manthey — APPROVED** [review]

**Pull request merged**